packages/kbn-esql-validation-autocomplete/src/definitions/types.ts

```diff
@@ -23,7 +23,22 @@ export interface FunctionDefinition {
       optional?: boolean;
       noNestingFunctions?: boolean;
       supportsWildcard?: boolean;
+      /**
+       * if set this indicates that the value must be a literal
+       * but can be any literal of the correct type
+       */
       literalOnly?: boolean;
+      /**
+       * if provided this means that the value must be one
+       * of the options in the array iff the value is a literal.
+       *
+       * String values are case insensitive.
+       *
+       * If the value is not a literal, this field is ignored because
+       * we can't check the return value of a function to see if it
+       * matches one of the options prior to runtime.
+       */
+      literalOptions?: string[];
```

Comment:
can we call it `values` to align with the `CommandDefinition` thing?

Reply:
Could you clarify a bit? I'm thinking options better describes what this is. "value" implies what the user actually chooses, whereas "options" says to me a set of values they are allowed to choose.